milestones/api.py

```diff
@@ -268,25 +284,36 @@ def add_course_content_milestone(course_key, content_key, relationship, mileston
         course_key=course_key,
         content_key=content_key,
         relationship=relationship,
-        milestone=milestone)
+        milestone=milestone,
+        requirement=requirement
+    )
 
 
-def get_course_content_milestones(course_key, content_key, relationship=None):
-    """
-    Retrieves the set of milestones for a given course content module
-    'relationship': optional filter on milestone relationship type (string, eg: 'requires')
-    Returns an array of dicts containing milestones
+def get_course_content_milestones(course_key=None, content_key=None, relationship=None, user=None):
```

Comment:
No, but there is for retrieving all CourseContentMilestone records for a given course. It seemed weird to enable that ability without enabling lookup without the other filters as well.

For gating we have a couple of cases where we need to find all CourseContentMilestones for a given course.
* Find all prerequisite subsections in a course
** https://github.com/edx/edx-platform/pull/11014/files#diff-b0c96d9aff40f6ecba9669e039b833a9R100
* Find all gated subsections in a course
** https://github.com/edx/edx-platform/pull/11014/files#diff-b0c96d9aff40f6ecba9669e039b833a9R221